[BE] Admin rights

nspc97 opened the pull request:
[BE] Admin rights
https://trello.com/c/5vF8vxMG/87-be-admin-rights

Commit:
[BE] Admin rights
https://trello.com/c/5vF8vxMG/87-be-admin-rights

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -275,7 +275,7 @@ public function getActivityDetails(Activity $activity): Response
         $rights = $this->accessRightsPolicy->canAccessActivity($activity, $user);
 
-        if ($rights === false) {
+        if ($rights === false && $user->getRoles() !== array('ROLE_ADMIN')) {
```
Use `$this->isGranted('ROLE_ADMIN');`

SpartakusMd replied:
Outside of controllers use `Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface::isGranted()`

Commit:
[BE] Job comments
https://trello.com/c/vRfRn8EZ/85-be-job-comments

Used $this->isGranted('ROLE_ADMIN');

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -685,8 +688,9 @@ public function applyForActivity(
+        $hasAccess = $this->isGranted('ROLE_ADMIN');
 
-        if ($activity->isPublic() === false) {
+        if ($hasAccess && $activity->isPublic() === false) {
```
Should be `if (!$hasAccess`

SpartakusMd commented on src/Entity/User.php:
```diff
@@ -210,6 +210,12 @@ class User implements UserInterface
 
+    /**
+     * The role of User
+     * @ORM\Column(type="string")
```
You can't save an array as string in database, use `@ORM\Column(type="json")`

SpartakusMd commented on src/Entity/User.php:
```diff
@@ -367,7 +373,12 @@ public function setUpdatedAt(DateTimeInterface $updatedAt): void
+        return array($this->roles);
+    }
+
+    public function setRoles(string $roles): void
```
`$roles` is an array, not string.

Commit:
[BE] Job comments
https://trello.com/c/vRfRn8EZ/85-be-job-comments
